src/codegen/compile.ml

```diff
@@ -1602,12 +1608,30 @@ module Tagged = struct
     assert (!Flags.gc_strategy = Flags.Incremental);
     1l
 
-  (* Note: Post allocation barrier must be applied after initialization *)
+  (* Note: post-allocation barrier must be applied after initialization *)
   let alloc env size tag =
+    assert (size > 1l);
     let name = Printf.sprintf "alloc_size<%d>_tag<%d>" (Int32.to_int size) (Int32.to_int (int_of_tag tag)) in
+    let overflow_mask n =
```

Comment:
Maybe it could be helpful to add a comment on how the page crossing logic works. The PR description is very nice.
As an intuition of that logic (if I am not mistaken), it is checking whether the new HP resides at the beginning of the page, i.e. the new address has leading zeros inside the page.

Reply:
added 80188facb